slash-commands/moderation/infractions.js

```diff
@@ -0,0 +1,80 @@
+const Discord = require('discord.js');
+const {SlashCommandBuilder} = require('@discordjs/builders');
+const {formatDistanceToNow} = require('date-fns');
+const {getEmbedHexFlairColor} = require('../../helpers/design');
+const {promisePool} = require('../../config/db');
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName('infractions')
+    .setDescription(
+      'Finds user infractions records in db and returns it to channel'
+    )
+    .addUserOption((option) =>
+      option
+        .setName('target')
+        .setDescription('The user (mention or id)')
+        .setRequired(true)
+    ),
+
+  async execute(interaction) {
+    try {
+      const targetUser = await interaction.options.getUser('target');
+      const infractions = await getInfractionsFromDB(targetUser);
+      await displayInfractionsLog(interaction, targetUser, infractions);
+    } catch (err) {
+      console.error(err);
+      interaction.reply(`There was an error reading the infractions.`);
+    }
+  },
+};
+
+async function getInfractionsFromDB(targetUser) {
+  try {
+    const sqlInfractions = `SELECT timestamp,reason,id,action,valid FROM infractions WHERE user = ?`;
+    const [infractions] = await promisePool.execute(sqlInfractions, [
+      targetUser.id,
+    ]);
+    return infractions;
+  } catch (err) {
+    throw new Error(err.message);
+  }
+}
+
+async function displayInfractionsLog(interaction, targetUser, infractions) {
+  const listOfinfractions = parseInfractions(interaction, infractions);
+  const totalInfractions = listOfinfractions.length;
+
+  if (totalInfractions) {
+    const infractionsEmbed = new Discord.MessageEmbed()
+      .setAuthor({
+        name: `${targetUser.username}#${targetUser.discriminator}'s infractions`,
+        iconURL: `https://cdn.discordapp.com/avatars/${targetUser.id}/${targetUser.avatar}.png`,
+      })
+      .setColor(getEmbedHexFlairColor())
+      .setDescription(`Total: ${totalInfractions}`)
+      .addFields(...listOfinfractions)
+      .setTimestamp()
+      .setFooter({text: interaction.guild.name});
+
+    return await interaction.reply({embeds: [infractionsEmbed]});
+  } else {
+    return await interaction.reply(
+      `${targetUser.username}#${targetUser.discriminator} doesn't have any infractions`
+    );
+  }
+}
+
+function parseInfractions(interaction, infractions) {
```

Comment:
Since the function parseInfractions doesn't need the interaction object we don't need to pass it to the function. I think only having one function parameter, infractions, would be better.

Reply:
I noticed that as well and ended up removing it in the records command PR when I refactored everything:
https://github.com/CodecademyCommunity/codecademy-discord-bot/pull/325/files#diff-adda48f9187fc6d9d0c1c4edc0c0ab2afe641731301ccdfe19bcd6d396de01bbR21

If you want I can remove it here, but it'll just get deleted in that PR anyway, so it might not be worth the extra rebasing.

Reply:
Ok. That sounds good.